atomix/cluster/src/main/java/io/atomix/raft/storage/RaftStorage.java

```diff
@@ -216,6 +216,11 @@ public void deleteSnapshotStore() {
    * @return The opened log.
    */
   public RaftLog openLog() {
+    final long lastWrittenIndex;
+    try (final MetaStore metaStore = openMetaStore()) {
```

Comment:
Why not pass the `lastWrittenIndex` instead of re-opening metastore?

Reply:
It's self-contained, everything we need is in the RaftStorage itself so I didn't see a reason to involve something else. It's also only done when opening the log so there weren't any performance or concurrency issues. But I can keep a reference at the RaftStorage and always return it, like a singleton